### PR TITLE
Implemented product details preview. 

### PR DESCRIPTION
**Pull request by ren-tham**

### CODE REVIEW

File: projects/storefrontlib/src/cms-components/product/product-tabs/product-details-tab/product-details-tab.component.html [hunk 1]
```diff
@@ -1,3 +1,20 @@
...
+    <div
+      *ngIf="!showMore; else fullContent"
+      [innerHTML]="product?.description.substring(0, 400)"
+    ></div>
+    <ng-template #fullContent
+      ><div [innerHTML]="product?.description"></div
+    ></ng-template>
+
+    <div *ngIf="product?.description.length > 400">
+      <a href="javascript:" *ngIf="showMore" (click)="toggle()">{{
```
Comment (WeizhengSap) [inline projects/storefrontlib/src/cms-components/product/product-tabs/product-details-tab/product-details-tab.component.html]: This is what we have in "product list". I think we can use similar buttons.
```
    <button
      *ngIf="state.maxVisible > state.topVisible"
      (click)="decreaseVisibleValues()"
      class="cx-action-link"
      cxFocus="moreorless"
    >
      {{ 'productList.showLess' | cxTranslate }}
    </button>

    <button
      *ngIf="state.maxVisible > 0 && state.maxVisible < facet.values.length"
      (click)="increaseVisibleValues()"
      class="cx-action-link"
      cxFocus="moreorless"
    >
      {{ 'productList.showMore' | cxTranslate }}
    </button>
```

File: projects/storefrontlib/src/cms-components/product/product-tabs/product-details-tab/product-details-tab.component.ts [hunk 2]
```diff
@@ -10,10 +10,14 @@ import { CurrentProductService } from '../../current-product.service';
 })
 export class ProductDetailsTabComponent implements OnInit {
   product$: Observable<Product>;
+  showMore: boolean;
```
Comment (WeizhengSap) [inline projects/storefrontlib/src/cms-components/product/product-tabs/product-details-tab/product-details-tab.component.ts]: better give it a default value: `showMore=false`

File: projects/storefrontlib/src/cms-components/product/product-tabs/product-details-tab/product-details-tab.component.html [hunk 3]
```diff
@@ -1,3 +1,20 @@
 <ng-container *ngIf="product$ | async as product">
-  <div class="container" [innerHTML]="product?.description"></div>
+  <div class="container">
+    <div
```
Comment (WeizhengSap) [inline projects/storefrontlib/src/cms-components/product/product-tabs/product-details-tab/product-details-tab.component.html]: You insert an additional `div` tag here. It's better to just keep it as 
```
<div class="container">
  ...the description
<div>

File: projects/storefrontlib/src/cms-components/product/product-tabs/product-details-tab/product-details-tab.component.html [hunk 4]
```diff
@@ -1,3 +1,20 @@
...
+      *ngIf="!showMore; else fullContent"
+      [innerHTML]="product?.description.substring(0, 400)"
+    ></div>
+    <ng-template #fullContent
```
Comment (WeizhengSap) [inline projects/storefrontlib/src/cms-components/product/product-tabs/product-details-tab/product-details-tab.component.html]: We can use `product.description.substr(0, 400)` no matter the description length longer than 400 or not, right?

**Comment by WeizhengSap** [comment]
Please consider to add it into e2e.

File: projects/storefrontlib/src/cms-components/product/product-tabs/product-details-tab/product-details-tab.component.html [hunk 5]
```diff
@@ -1,3 +1,20 @@
 <ng-container *ngIf="product$ | async as product">
-  <div class="container" [innerHTML]="product?.description"></div>
+  <div class="container">
+    <div
+      *ngIf="!showMore; else fullContent"
+      [innerHTML]="product?.description.substring(0, 400)"
```
Comment (WeizhengSap) [inline projects/storefrontlib/src/cms-components/product/product-tabs/product-details-tab/product-details-tab.component.html]: it's better to create a const for this `400`.

File: projects/storefrontlib/src/cms-components/product/product-tabs/product-details-tab/product-details-tab.component.ts (hunk 2 above)
Comment (ren-tham) [reply projects/storefrontlib/src/cms-components/product/product-tabs/product-details-tab/product-details-tab.component.ts]: done

File: projects/storefrontlib/src/cms-components/product/product-tabs/product-details-tab/product-details-tab.component.html (hunk 1 above)
Comment (ren-tham) [reply projects/storefrontlib/src/cms-components/product/product-tabs/product-details-tab/product-details-tab.component.html]: done

File: projects/storefrontlib/src/cms-components/product/product-tabs/product-details-tab/product-details-tab.component.html (hunk 4 above)
Comment (ren-tham) [reply projects/storefrontlib/src/cms-components/product/product-tabs/product-details-tab/product-details-tab.component.html]: Yes.

File: projects/storefrontlib/src/cms-components/product/product-tabs/product-details-tab/product-details-tab.component.html (hunk 5 above)
Comment (ren-tham) [reply projects/storefrontlib/src/cms-components/product/product-tabs/product-details-tab/product-details-tab.component.html]: done

**Pull request closed without merging**